chain/chain/src/chain.rs

```diff
@@ -2976,14 +2892,27 @@ impl<'a> ChainUpdate<'a> {
             return Err(ErrorKind::Other("Invalid block".into()).into());
         }
 
+        let protocol_version =
+            self.runtime_adapter.get_epoch_protocol_version(&block.header().epoch_id())?;
         if !block.verify_gas_price(
             prev_gas_price,
-            self.block_economics_config.min_gas_price,
-            self.block_economics_config.max_gas_price,
-            self.block_economics_config.gas_price_adjustment_rate,
+            self.block_economics_config.min_gas_price(protocol_version),
+            self.block_economics_config.max_gas_price(protocol_version),
+            self.block_economics_config.gas_price_adjustment_rate(protocol_version),
         ) {
-            byzantine_assert!(false);
-            return Err(ErrorKind::InvalidGasPrice.into());
+            let prev_protocol_version =
+                self.runtime_adapter.get_epoch_protocol_version(&prev_epoch_id)?;
+            let is_gas_price_valid = if prev_protocol_version != protocol_version
+                && protocol_version == MIN_PROTOCOL_VERSION_NEP_92
+            {
+                block.header().gas_price() == MIN_GAS_PRICE_NEP_92
+            } else {
+                false
+            };
```

Comment:
This logic should not be needed. `verify_gas_price` compares the gas price to `compute_new_gas_price`, which cannot return anything smaller than the passed in `min_gas_price` (unless the `max_gas_price` is smaller than the `min_gas_price`, which would be a bug)

Reply:
No. verify_gas_price verifies that gas price in the current block is valid compared to gas price in the previous block. So if this is the first block of the epoch where the gas price changes, we have to special case it.

Reply:
Verify gas price compares the gas stores in the block to the expected:

```
    pub fn verify_gas_price(
        &self,
        prev_gas_price: Balance,
        min_gas_price: Balance,
        max_gas_price: Balance,
        gas_price_adjustment_rate: Rational,
    ) -> bool {
        let gas_used = Self::compute_gas_used(self.chunks(), self.header().height());
        let gas_limit = Self::compute_gas_limit(self.chunks(), self.header().height());
        let expected_price = Self::compute_new_gas_price(
            prev_gas_price,
            gas_used,
            gas_limit,
            gas_price_adjustment_rate,
            min_gas_price,
            max_gas_price,
        );
        self.header().gas_price() == expected_price
    }
```

The expected is computed as 

```
    pub fn compute_new_gas_price(
        prev_gas_price: Balance,
        gas_used: Gas,
        gas_limit: Gas,
        gas_price_adjustment_rate: Rational,
        min_gas_price: Balance,
        max_gas_price: Balance,
    ) -> Balance {
        if gas_limit == 0 {
            prev_gas_price
        } else {
            let numerator = 2 * *gas_price_adjustment_rate.denom() as u128 * u128::from(gas_limit)
                - *gas_price_adjustment_rate.numer() as u128 * u128::from(gas_limit)
                + 2 * *gas_price_adjustment_rate.numer() as u128 * u128::from(gas_used);
            let denominator =
                2 * *gas_price_adjustment_rate.denom() as u128 * u128::from(gas_limit);
            let new_gas_price =
                U256::from(prev_gas_price) * U256::from(numerator) / U256::from(denominator);
            if new_gas_price > U256::from(max_gas_price) {
                max_gas_price
            } else {
                max(new_gas_price.as_u128(), min_gas_price)
            }
        }
    }
```

Note the `max(new_gas_price.as_u128(), min_gas_price)` at the very end. The `expected` gas price will be computed to be the min gas price in the epoch, which will be `MIN_GAS_PRICE_NEP_92`.